src/modal/modal.js

```diff
@@ -261,18 +249,21 @@ angular.module('ui.bootstrap.modal', [])
       }
 
       function removeAfterAnimate(domEl, scope, done) {
-        // Closing animation
-        scope.animate = false;
-
-        if (domEl.attr('modal-animation') && $animate.enabled()) {
-          // transition out
-          domEl.one('$animate:close', function closeFn() {
-            $rootScope.$evalAsync(afterAnimating);
-          });
-        } else {
-          // Ensure this call is async
-          $timeout(afterAnimating);
-        }
+        var closingDeferred;
```

Comment:
Would it make sense to do
```
var resolveFn;
var closingPromise = $q(function(resolve) {
  resolveFn = resolve;
});
```
And change setIsAsync to
```
function setIsAsync() {
  return function() {
    resolveFn();
  };
}
```
Seems like this would be potentially less prone to error with ensuring `closingPromise` exists.